### ISSUE TITLE
Saving content to contenttype sometimes results in "Call to a member function isEmpty() on array" Exception

### ISSUE THREAD

**Issue by surips** [bolt/core#1285]
When trying to save new content to my contenttype or when trying to access the relations I sometimes get an "Call to a member function isEmpty() on array"

Details
-------

| Question                      | Answer
|----------------- | ------
| Relevant Bolt Version | 4.0 RC15
| Install type                   |  Composer install
| PHP version                 | 7.4 
| Web server                  | Apache / Nginx 

Reproduction
------------

### Bug summary

My contenttype (see below) has a relation to another contenttype (businessareas). Before updating to RC15 everything seemed normal. I filled my contenttype with content and created relations to the other contenttype and had no problems. But now, if I want to change something and click the "save" button the exception that you see on the screenshots occurs. It does not occur with every instance of this contenttype.

### Specifics

I have the following contenttype:

```
usecases:
    name: Use Cases
    slug: usecases
    singular_name: Use Case
    singular_slug: usecase
    fields:
        title:
           type: text
           localize: true
        slug:
           type: slug
           uses: title
        teaserimage:
           type: image
           extensions: [gif, jpg, png]
           localize: true
        image:
           type: image
           extensions: [gif, jpg, png]
           localize: true
        headline:
            type: text
            localize: true
        teaser:
           type: html
           localize: true
        content:
           type: html
           localize: true
        isReference:
           type: checkbox
        companyname:
           type: text
           localize: true
        companyurl:
           type: text
           localize: true
        companylogo:
           type: image
           localize: true
    locales: ['de','en']
    relations:
        businessareas:
            multiple: false
            order: -id
```

See screenshots below for exception:
![Screenshot_2020-04-08 Call to a member function isEmpty() on array (500 Internal Server Error)(1)](https://user-images.githubusercontent.com/60933485/78779641-8ec01200-799d-11ea-8247-008533fa84a9.png)
![Screenshot_2020-04-08 Call to a member function isEmpty() on array (500 Internal Server Error)(2)](https://user-images.githubusercontent.com/60933485/78779650-9089d580-799d-11ea-89ca-dab4dd4e649f.png)

When using ` {{dump(record|related())}}` in a for-loop that iterates over all usecases the same exception occurs. Other fields of the contenttype can be dumped without a problem.

### Steps to reproduce
1. Define contenttype as mentioned above
2. Save new content

### Expected result

Being able to save content normally.

### Actual result

The aforementioned exception: Call to a member function isEmpty() on array

**Comment by bobdenotter** [comment]
Hi @surips, 

Thanks for the extensive report. This issue should be fixed in this PR: #1289

I'll package a new "RC 16" this afternoon, which will have the fix. Let me know if you can confirm it's fixed afterwards. :-)

**Comment by surips** [comment]
@bobdenotter  Thanks, RC16 seems to have solved the problem! :-)
